eng/pipelines/code-quality-reports.yml

```diff
@@ -5,136 +5,147 @@ parameters:
-          mavenOptions: '$(MemoryOptions) $(LoggingOptions)'
-          javaHomeOption: 'JDKVersion'
-          jdkVersionOption: $(JavaBuildVersion)
-          jdkArchitectureOption: 'x64'
-          publishJUnitResults: false
-        condition: and(succeeded(), ne(variables['LintingGoals'], ''))
-
-      - task: Maven@3
-        displayName: 'Run Linting'
-        inputs:
-          mavenPomFile: ClientFromSourcePom.xml
-          goals: $(LintingGoals)
-          # Skip failing on Checkstyle, RevApi, and Spotbugs errors.
-          # This is being done as the error reporting will be captured, so if any error reports exist the build can be
-          # failed later allowing all libraries to be validated.
-          options: '$(DefaultOptions) -T 2C -Dcheckstyle.failOnViolation=false -Dcheckstyle.failsOnError=false -Dspotbugs.failOnError=false -Drevapi.failBuildOnProblemsFound=false -pl "!com.azure:azure-sdk-from-source,$(SkipLintingProjects)"'
-          mavenOptions: '$(MemoryOptions) $(LoggingOptions)'
-          javaHomeOption: 'JDKVersion'
-          jdkVersionOption: $(JavaBuildVersion)
-          jdkArchitectureOption: 'x64'
-          publishJUnitResults: false
-        condition: and(succeeded(), ne(variables['LintingGoals'], ''))
-
-      - task: PowerShell@2
-        inputs:
-          filePath: $(System.DefaultWorkingDirectory)/eng/pipelines/scripts/Get-Linting-Reports.ps1
-          arguments: -StagingDirectory $(Build.ArtifactStagingDirectory)
-          pwsh: true
-          workingDirectory: $(Pipeline.Workspace)
-        displayName: 'Generate Linting Reports'
-        condition: and(always(), ne(variables['LintingGoals'], ''))
-
-      - template: /eng/common/pipelines/templates/steps/publish-1es-artifact.yml
-        parameters:
-          ArtifactName: linting-report
-          ArtifactPath: '$(Build.ArtifactStagingDirectory)/troubleshooting/linting-report.zip'
-          CustomCondition: eq(variables['HAS_TROUBLESHOOTING'], 'true')
-
-      - pwsh: |
-          Write-Host "Linting errors have been found, download the 'linting-report.zip' artifact to see what was found."
-          exit 1
-        displayName: 'Fail Pipeline if Linting Reports Exist'
-        condition: eq(variables['HAS_TROUBLESHOOTING'], 'true')
-
-      - task: PowerShell@2
-        displayName: Clean up maven local cache
-        inputs:
-          pwsh: true
-          filePath: $(Build.SourcesDirectory)/eng/scripts/Remove-Azure-Artifacts-From-Cache.ps1
-          arguments: >
-            -MavenCacheFolder $(MAVEN_CACHE_FOLDER)
-        continueOnError: true
+  - name: oneESTemplateTag
+    type: string
+    default: release
+
+extends:
+  template: /eng/pipelines/templates/stages/1es-redirect.yml
+  parameters:
+    oneESTemplateTag: ${{ parameters.oneESTemplateTag }}
```

Comment:
This can be removed we only use it for template as a canary validation pipeline.